cvrf2csaf/cvrf2csaf.py

```diff
@@ -28,30 +42,85 @@ class DocumentHandler:
+            elif tag == 'DocumentType':
+                # Map DocumentType --> /document/category/
+                self.document_category.create_csaf(root_element=elem)
+            elif tag == 'DocumentDistribution':
+                self.document_distribution.create_csaf(root_element=elem)
+            elif tag == 'DocumentNotes':
+                self.document_notes.create_csaf(root_element=elem)
+            elif tag == 'DocumentPublisher':
+                self.document_publisher.create_csaf(root_element=elem)
+            elif tag == 'DocumentReferences':
+                self.document_references.create_csaf(root_element=elem)
+            elif tag == 'DocumentTitle':
+                self.document_title.create_csaf(root_element=elem)
             elif tag == 'DocumentTracking':
-                self.document_tracking.create_csaf(elem)
-            elif tag == 'ToDo':
-                # ToDo: Going through it tag by tag for further parsing
-                pass
+                self.document_tracking.create_csaf(root_element=elem)
+            elif tag == 'ProductTree':
+                self.product_tree.create_csaf(root_element=elem)
+            elif tag == 'Vulnerability':
+                self.vulnerability.create_csaf(root_element=elem)
+            elif tag in ['comment']:
+                self.comment.create_csaf(root_element=elem)
             else:
                 logging.warning(f'Not handled input tag {tag}. No parser available.')
 
     def _compose_final_csaf(self) -> dict:
+        # Merges first level leaves into final CSAF document.
+        # [mapping table](https://github.com/tschmidtb51/csaf/blob/csaf-2.0-what-is-new-table/notes/whats-new-csaf-v2.0-cn01.md#e4-mapped-elements)
+
         final_csaf = {'document': {}}
+        final_csaf['document']['acknowledgments'] = self.document_acknowledgments.csaf
+        final_csaf['document']['aggregate_severity'] = self.document_aggregate_severity.csaf
+        final_csaf['document']['category'] = self.document_category.csaf
+        final_csaf['document']['distribution'] = self.document_distribution.csaf
+        final_csaf['document']['lang'] = self.document_lang.csaf
+        final_csaf['document']['notes'] = self.document_notes.csaf
         final_csaf['document']['publisher'] = self.document_publisher.csaf
+        final_csaf['document']['source_lang'] = self.document_source_lang.csaf
+        final_csaf['document']['references'] = self.document_references.csaf
+        final_csaf['document']['title'] = self.document_title.csaf
         final_csaf['document']['tracking'] = self.document_tracking.csaf
+        final_csaf['document']['product_tree'] = self.product_tree.csaf
+
+        final_csaf['document']['tracking'] = self.document_tracking.csaf
+        final_csaf['document']['references'] = self.document_references.csaf
+        final_csaf['document']['product_tree'] = self.product_tree.csaf
+        final_csaf['document']['vulnerability'] = self.vulnerability.csaf
```

Comment:
The `vulnerabilities` is not on the same level as `/document/tracking`. `vulnerabilities` is on the same level as `document`. Also it is called `vulnerabilities` not `vulnerability`.